README.md

```diff
@@ -87,25 +85,79 @@ src/Dhall/TH.hs:196: toNestedHaskellType
 (Please note these warnings are just for demonstration and not necessarily weeds
 in the Dhall project).
 
+## Configuration options
+
+| Name             | Default value                        | Description |
+| ---------------- | ------------------------------------ | --- |
+| roots            | `[ "Main.main", "^Paths_weeder.*" ]` | Any declarations matching these regular expressions will be considered as alive. |
+| type-class-roots | `false`                              | Consider all instances of type classes as roots. Overrides `root-classes` and `root-instances`. |
+| root-classes     | `[ "IsString", "IsList" ]`           | Instances in matching modules of matching classes will be considered as roots. Note that this does not mark the type class declarations themselves as roots, so a type class with no instances may still show up as a weed. |
+| root-instances   | `[]`                                 | Instances in matching modules with matching types will be considered as roots. |
+| unused-types     | `false`                              | Enable analysis of unused types. |
+
+`root-classes` and `root-instances` accept both TOML tables and string literals. String
+literals are interpreted as leaving `module` unspecified, therefore matching on all 
+modules. See the following example from the test suite:
+
+``` toml
+root-instances = [ { module = "Spec.ConfigInstanceModules.Module1", instance = "Bounded T" }
+                 , "Read T" 
+                 , { module = "Spec.ConfigInstanceModules.Module3" }
+                 ]
+
+root-classes = [ { class = "Enum" }
+               , { module = "Spec.ConfigInstanceModules.Module2", class = "Show" } 
+               ]
+```
+
+## Exit codes
+
+Weeder emits the following failing exit codes:
+
+| Exit code | Cause |
+| --- | --- |
+| 228 | One or more weeds found |
+|  1  | Generic failing exit code |
+|  2  | Failure to read HIE file due to GHC version mismatch |
+|  3  | Failure to parse config file |
+|  4  | No HIE files found |
+
 # Tips
 
 - You may want to add `^Paths_.*` to the roots in `weeder.toml` to ignore the
   `Paths_packageName` module automatically generated by Cabal.
 
+- You can automatically write and use a default configuration file by calling 
+  Weeder with the `--write-default-config` flag, if no configuration file is
+  found.
+
+- You can mandate explicitly specifying every option in the configuration by 
+  calling Weeder with the `--no-default-fields` flag. This can prevent being
+  caught off guard by new configuration options or changes to default values.
+
+- To mark all instances in a module `M` as roots, add `{ module = "^M$" }`
+  to either `root-classes` or `root-instances`.
```

Comment:
Hmm, either? Surely just `root-instances`?

Reply:
`module` does the same thing in both `root-classes` and `root-instances`. I'll amend this PR for #136 later.